cfn-resources/database-user/cmd/resource/resource.go

```diff
@@ -19,17 +17,23 @@ func setup() {
 	util.SetupLogger("mongodb-atlas-database-user")
 }
 
-// validateModel inputs based on the method
-func validateModel(event constants.Event, model *Model) *handler.ProgressEvent {
-	return validator.ValidateModel(event, validation.ModelValidator{}, model)
+// function to validate inputs to all actions
+func validateModel(fields []string, model *Model) *handler.ProgressEvent {
+	return validator.ValidateModel(fields, model)
 }
 
+var CreateRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "DatabaseName", "ProjectId", "Roles", "Username"}
+var ReadRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId", "DatabaseName", "Username"}
+var UpdateRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "DatabaseName", "ProjectId", "Roles", "Username"}
+var DeleteRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId", "DatabaseName", "Username"}
+var ListRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId"}
```

Comment:
Mentioned this in 
but we have almost the same logic in https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/project/cmd/validation/validate.go

Would it be possible to reuse that?

Reply:
We have different fields required for each methods in different resources.
For Example,
Required Fields for Read Method in Database-User: {"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ProjectId", **"DatabaseName", "Username"**}
Required Fields for Read Method in Cloud-Backup: {"ProjectId", **"ClusterName", "Id",** "ApiKeys.PublicKey", "ApiKeys.PrivateKey"}

Reply:
Cool, thank you for clarifying!